internal/agent/agent.go

```diff
@@ -104,31 +107,59 @@ func RunAgent(
 		go func() {
 			err := server.ServeSSE(ctx, l)
 			cancel(err)
+			_ = l.Close()
 		}()
 		b := env.NewStreamBuilder()
 		b.SetHTTPMux(&gMux{m: mux, prefix: "/" + name})
 		b.SetLogger(logger)
 		b.SetEnvVarLookupFunc(func(key string) (string, bool) {
 			return envVarLookupFunc(context.Background(), key)
 		})
-		if !agent.Input.IsZero() {
-			input, _ := yaml.Marshal(agent.Input)
-			if err := b.AddInputYAML(string(input)); err != nil {
-				return nil, fmt.Errorf("failed to add agent input: %w", err)
-			}
+		configs := []struct {
+			name    string
+			node    yaml.Node
+			builder func(string) error
+		}{
+			{
+				name:    "input",
+				node:    agent.Input,
+				builder: b.AddInputYAML,
+			},
+			{
+				name:    "output",
+				node:    agent.Output,
+				builder: b.AddOutputYAML,
+			},
+			{
+				name:    "metrics",
+				node:    agent.Metrics,
+				builder: b.SetMetricsYAML,
+			},
+			{
+				name:    "logger",
+				node:    agent.Logger,
+				builder: b.SetLoggerYAML,
+			},
+			{
+				name:    "tracer",
+				node:    agent.Tracer,
+				builder: b.SetTracerYAML,
+			},
```

Comment:
Do we also need something for [`processor`](https://pkg.go.dev/github.com/redpanda-data/benthos/v4@v4.50.0/public/service#StreamBuilder.AddProcessorYAML) and [`rate_limit`](https://pkg.go.dev/github.com/redpanda-data/benthos/v4@v4.50.0/public/service#StreamBuilder.AddRateLimitYAML)? Maybe [resources](https://pkg.go.dev/github.com/redpanda-data/benthos/v4@v4.50.0/public/service#StreamBuilder.AddResourcesYAML) too?

Reply:
And probably buffering too... We'll get there I guess :)